chain/src/view.rs

```diff
@@ -17,7 +18,7 @@ use crate::{
 /// initialized, so they will error on an empty state.
 //#[async_trait(?Send)]
 #[async_trait]
-pub trait StateReadExt: StateRead {
+pub trait StateReadExt: StateReadProto {
```

Comment:
Do all these bounds have to change? Since `StateReadProto` has a blanket implementation (or should), it should be present for any `StateRead`.

Reply:
good catch thanks, yeah they have blanket impls!